dask/backends.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from functools import lru_cache
+from typing import Callable
+
+from dask import config
+from dask.compatibility import entry_points
+
+
+class DaskBackendEntrypoint(ABC):
+    """Base Collection-Backend Entrypoint Class
+
+    Most methods in this class correspond to collection-creation
+    for a specific library backend. Once a collection is created,
+    the existing data will be used to dispatch compute operations
+    within individual tasks. The backend is responsible for
+    ensuring that these data-directed dispatch functions are
+    registered when ``__init__`` is called.
+    """
+
+    @abstractmethod
+    def __init__(self):
+        """Register data-directed dispatch functions
+
+        This may be a ``pass`` operation if the data dispatch functions
+        are already registered within the same module that the
+        ``DaskBackendEntrypoint`` subclass is defined.
+        """
+        raise NotImplementedError
+
+
+@lru_cache(maxsize=1)
+def detect_entrypoints():
+    entrypoints = entry_points("dask.backends")
+    return {ep.name: ep for ep in entrypoints}
+
+
+class CreationDispatch:
+    """Simple backend dispatch for collection-creation functions"""
+
+    def __init__(self, config_field, default, name=None):
+        self._lookup = {}
+        self.config_field = config_field
+        self.default = default
+        if name:
+            self.__name__ = name
+
+    def register_backend(self, backend: str, cls_target: DaskBackendEntrypoint):
+        """Register a target class for a specific backend label"""
+        raise NotImplementedError
```

Comment:
Annotate the return type (and require that the backend instance is returned)?

```
from typing import TypeVar

T = TypeVar("T", bound=DaskBackendEntrypoint)

...
def register_backend(self, backend: str, instance: T) -> T:
    ...
```

Then we can simplify `dispatch` a little